yt_dlp/options.py

```diff
@@ -410,12 +410,20 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
             '("-" for stdin). Can be used multiple times and inside other configuration files'))
     general.add_option(
         '--plugin-dirs',
-        dest='plugin_dirs', metavar='PATH', action='append',
+        metavar='PATH',
+        dest='plugin_dirs',
+        action='append',
```

Comment:
`append` mutates the `default` list, which is a problem because we reuse the parser multiple times. This is why you are getting repeated entries and having to use `set`. Our callback instead handles this correctly.
```suggestion
        action='callback',
        callback=_list_from_options_callback,
        type='str',
        callback_kwargs={'delim': None},
```

Reply:
Ah, neat. That also fixes the issue of supplying multiple `--no-plugin-dirs`, e.g. `--no-plugin-dirs --plugin-dirs /xyz --no-plugin-dirs --plugin-dirs /new/path`

I was copying from --config-locations which suffers from these issues

Reply:
> I was copying from --config-locations which suffers from these issues

Good catch. Though in its case, this'll do

```diff
diff --git a/yt_dlp/options.py b/yt_dlp/options.py
index 50341695b..b2079dc7e 100644
--- a/yt_dlp/options.py
+++ b/yt_dlp/options.py
@@ -398,7 +398,7 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
             '(Alias: --no-config)'))
     general.add_option(
         '--no-config-locations',
-        action='store_const', dest='config_locations', const=[],
+        action='store_const', dest='config_locations', const=None,
         help=(
             'Do not load any custom configuration files (default). When given inside a '
             'configuration file, ignore all previous --config-locations defined in the current file'))
```

Reply:
Committed